include/deal.II/lac/scalapack.h

```diff
@@ -0,0 +1,501 @@
+   */
+  ScaLAPACKMatrix<NumberType> &
+  operator = (const FullMatrix<NumberType> &);
+
+  /**
+   * Copy the content of the distributed matrix into @p matrix.
+   *
+   * @note This function should only be used for relatively small matrix
+   * dimensions. It is primarily intended for debug purposes.
+   */
+  void copy_to (FullMatrix<NumberType> &matrix) const;
+
+  /**
+   * Compute the Cholesky factorization of the matrix using ScaLAPACK
+   * function pXpotrf. The result of factorization is stored in this object.
+   */
+  void compute_cholesky_factorization ();
+
+  /**
+   * Invert the matrix by first computing Cholesky factorization and then
+   * building the actual inverse using pXpotri. The inverse is stored
+   * in this object.
+   */
+  void invert();
+
+  /**
+   * Compute all eigenvalues of real symmetric matrix using pXsyev.
+   * After successful computation the eigenvalues are returned in @p eigenvalues
+   * in ascending order.
+   */
+  void eigenvalues_symmetric (std::vector<NumberType> &eigenvalues);
+
+  /**
+   * Compute all eigenpairs of real symmetric matrix using pXsyev.
+   * After successful computation the eigenvalues are returned in @p eigenvalues
+   * in ascending order. The eigenvectors are stored in the columns of the matrix,
+   * therefore overwriting the original content of the matrix.
+   */
+  void eigenpairs_symmetric (std::vector<NumberType> &eigenvalues);
+
+  /**
+   * Estimate the the condition number of a SPD matrix in $l_1$-norm.
+   * The matrix has to be in Cholesky state. The reciprocal of the
+   * condition number is returned in order to avoid the possibility of
+   * overflow when the condition number is very large.
+   *
+   * @p a_norm shall contain $l_1$-norm of the matrix prior to calling
+   * Cholesky factorization.
+   *
+   * @note an alternative is to compute the inverse of the matrix
+   * explicitly and manually constructor $k_1 = ||A||_1 ||A^{-1}||_1$.
+   */
+  NumberType reciprocal_condition_number(const NumberType a_norm) const;
+
+  /**
+   * Compute $l_1$-norm of the matrix.
+   * @return
+   */
+  NumberType l1_norm() const;
+
+  /**
+   * Compute $l_{\infty}$ norm of the matrix.
+   */
+  NumberType linfty_norm() const;
+
+  /**
+   * Compute the Frobenius norm of the matrix.
+   */
+  NumberType frobenius_norm() const;
+
+  /**
+   * Number of rows of the matrix $M \times N$.
+   */
+  int m() const;
+
+  /**
+   * Number of columns of the matrix $M \times N$.
+   */
+  int n() const;
+
+  /**
+   * Number of local rows on this MPI processes.
+   */
+  int local_m() const;
+
+  /**
+   * Number of local columns on this MPI process.
+   */
+  int local_n() const;
+
+  /**
+   * Return global row number for the given local row @p loc_row .
+   */
+  int global_row(const int loc_row) const;
+
+  /**
+   * Return global column number for the given local column @p loc_column.
+   */
+  int global_column(const int loc_column) const;
+
+  /**
+   * Read access to local element.
```

Comment:
What does this return? The value for the given locally-owned element?

Reply:
correct.